src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -510,16 +551,16 @@ def work_for_vts_rsc_jar_library(vts, ctx):
           args = [
             '--include-scala-library-synthetics',
           ] + args
+        distribution = self._get_jvm_distribution()
         metacp_wu = self._runtool(
           'scala.meta.cli.Metacp',
           'metacp',
           args,
           distribution,
           tgt=tgt,
-          input_files=tuple(dependency_classpath + scalac_classpath_path_entries + classpath_rel),
+          input_files=tuple(metacp_dependencies + classpath_rel),
           output_dir=rsc_index_dir)
-        metacp_stdout = stdout_contents(metacp_wu)
-        metacp_result = json.loads(metacp_stdout)
+        metacp_result = json.loads(stdout_contents(metacp_wu))
```

Comment:
Looks like `rsc_compile.py` is gradually turning into a wall of text (it's close to 1kloc right now). I don't know about coding conventions in this project, so feel free to ignore this request, but perhaps it would be beneficial to split this logic into multiple modules (e.g. by splitting the rsc_compile task into multiple tasks such as metacp, rsc and scalac)?

Reply:
Breaking this up is on my todo list. One of the issues is that I don't want to break it up into separate v1 style tasks because I don't think that buys much.

I'm planning on doing a refactoring pass on it once I have a good feel for the bones of the abstractions that are needed, but I want to leave it a bit too verbose at the moment.